openquake/calculators/event_based_risk.py

```diff
@@ -312,12 +307,18 @@ def post_execute(self, result):
                 elif cb.user_provided:  # risk curves
                     # data is a list of dicts asset idx -> counts
                     poes = cb.build_poes(N, data, ses_ratio)
-                    self.datasets[o, l, r].dset[:] = poes
+                    if o == RC:
+                        rcurves[lt][:, r] = poes
+                    elif insured_losses:
+                        icurves[lt][:, r] = poes
                     saved[self.outs[o]] += poes.nbytes
                 self.datastore.hdf5.flush()
 
         self.datastore['avg_losses-rlzs'] = avg_losses
         saved['avg_losses-rlzs'] = avg_losses.nbytes
+        self.datastore['rcurves-rlzs'] = rcurves
```

Comment:
No, this is fine, rcurves are always generated, the optional curves are the icurves.